docs/DeveloperGuide.md

```diff
@@ -1,38 +1,726 @@
+
+#### Usage: `saveData()`
+
+To illustrate the overall flow on how to save data, refer to the sequence diagram below.
+
+The general idea is to `tokenize` it first into the `ArrayList<String>` format calling before
+calling `Storage`'s `saveData` method.
+
+![](./diagrams/StorageSequence-Saving_Data__Emphasis_on_Storage_Subroutine_.png)
+
+#### Design Considerations
+
+- Only filenames defined by public string constants in the `Storage` class.
+  This is meant to prevent developer mis-use and control what exactly waht files WellNUS++ can create.
+- Internally, each entry in `ArrayList<String>` will be delimited by ` --\n`, where \n is `System.LineSeparator()`.
+  This was chosen due to the invariant property of `' --'` in the context of WellNUS++. Due to the way all user input
+  is filtered by the `CommandParser`, the chosen delimiter should never show up in any data input, such as a habit name
+  from `AtomicHabits`
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+* NUS Computing and Engineering students
+* Spend lots of time coding on their IDE and type relatively fast
+* Have to regularly use digital gadgets and Internet for their courses
+* Very familiar with command line interfaces
+* Stressed about academy and many others
+* Busy with work and drowning in deadlines
+* Wants to improve their wellness
+* Sometimes unmotivated with short attention span
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+NUS Computing and Engineering students are often busy with work and sometimes will neglect their wellness. This app aims
+to help NUS Computing and Engineering students improve their overall wellness by encouraging the **cultivation of
+meaningful
+atomic habits**, **practice of self reflection** and **usage of offline timer to stay focused**. By using this app,
+we hope users will be more aware of the healthiness of their daily life and take actions to improve their wellness.<br>
+<br>
+WellNUS++ is a CLI app, primarily due to the following reasons:
+
+* Computing students generally type fast and prefer typing to mouse due to their daily coding routines.
+* Due to the data heavy nature and personalised user input of this app, typing will be preferred to clicking.
+* In particular, our application is built to reduce context switching. Users can launch the application from the comfort
+  of their favourite IDE’s terminal to reduce disruption to their daily coding lives.
+* Instead of using electronics with fancy GUI, this CLI app gives computing students an opportunity to minimise digital
+  interaction which will be beneficial for their wellness.
+* The app is gamified to make it more attractive for students to use. Levels and micro-goals incentivise our
+  users to keep using the app’s features, allowing them to focus on their work and achieve wellness.
 
 ## User Stories
```

Comment:
User stories language format could be changed to fit the "I want to ..." sentence. For e.g. I want to use keyboard instead of mouse vs I want to I can use keyboard instead of mouse